File: src/main/java/com/github/sorusclient/client/adapter/v1_8_9/LivingEntityImpl.java
package com.github.sorusclient.client.adapter.v1_8_9;

import com.github.sorusclient.client.adapter.IItem;
import com.github.sorusclient.client.adapter.ILivingEntity;
import com.github.sorusclient.client.adapter.IPotionEffect;
import v1_8_9.net.minecraft.client.MinecraftClient;
import v1_8_9.net.minecraft.entity.Entity;
import v1_8_9.net.minecraft.entity.LivingEntity;
import v1_8_9.net.minecraft.entity.effect.StatusEffectInstance;
import v1_8_9.net.minecraft.item.ItemStack;

import java.util.ArrayList;
import java.util.List;

public class LivingEntityImpl extends EntityImpl implements ILivingEntity {

    public LivingEntityImpl(Entity entity) {
        super(entity);
    }

    @Override
    public List<IPotionEffect> getEffects() {
        List<IPotionEffect> effects = new ArrayList<>();
        for (StatusEffectInstance effect : ((LivingEntity) this.entity).method_6120()) {
            effects.add(new PotionEffectImpl(effect));
        }
        return effects;
    }

    @Override
    public List<IItem> getArmor() {
        List<IItem> armor = new ArrayList<>();
        for (ItemStack itemStack : this.entity.getArmorStacks()) {
            if (itemStack != null) {
                armor.add(0, new ItemImpl(itemStack));
            }
        }
        return armor;
    }

    @Override
    public double getHealth() {
        return ((LivingEntity) this.entity).getHealth();
    }

    @Override
    public double getMaxHealth() {
        return ((LivingEntity) this.entity).getMaxHealth();
    }

    @Override
    public double getAbsorption() {
        return ((LivingEntity) this.entity).getAbsorption();
    }

}

File: src/main/kotlin/com/github/sorusclient/client/adapter/v1_18_1/LivingEntityImpl.kt
package com.github.sorusclient.client.adapter.v1_18_1

import com.github.sorusclient.client.adapter.IItem
import com.github.sorusclient.client.adapter.ILivingEntity
import com.github.sorusclient.client.adapter.IPotionEffect
import v1_18_1.net.minecraft.entity.Entity
import v1_18_1.net.minecraft.entity.LivingEntity

open class LivingEntityImpl(entity: Entity) : EntityImpl(entity), ILivingEntity {

    override val effects: List<IPotionEffect>
        get() {
            val effects: MutableList<IPotionEffect> = ArrayList()
            for (effect in (entity as LivingEntity).statusEffects) {
                effects.add(PotionEffectImpl(effect))
            }
            return effects
        }

    override val armor: List<IItem?>
        get() {
            val armor: MutableList<IItem?> = ArrayList()
            for (itemStack in entity.armorItems) {
                if (itemStack != null) {
                    armor.add(0, ItemImpl(itemStack))
                } else {
                    armor.add(0, null)
                }
            }
            return armor
        }

    override val health: Double
        get() = (entity as LivingEntity).health.toDouble()

    override val maxHealth: Double
        get() = (entity as LivingEntity).maxHealth.toDouble()

    override val absorption: Double
        get() = (entity as LivingEntity).absorptionAmount.toDouble()

}